### PR TITLE
Move loading-icon logic into a loading-icon component

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: apps/foundation-demo/src/app/core/loading-icon/loading-icon/loading-icon.component.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Component, ViewEncapsulation } from '@angular/core';
+
+@Component({
+  selector: '[loading-icon]',
+  templateUrl: './loading-icon.component.html',
+  styleUrls: ['./loading-icon.component.scss'],
+  host: {'class': 'loading-icon'},
+  encapsulation: ViewEncapsulation.None
+})
```
Comment (damienwebdev) [inline apps/foundation-demo/src/app/core/loading-icon/loading-icon/loading-icon.component.ts]: Can we change the `changeDetection` strategy to `OnPush` for better perf?

File: apps/foundation-demo/src/app/core/loading-icon/loading-icon/loading-icon.component.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Component, ViewEncapsulation } from '@angular/core';
+
+@Component({
+  selector: '[loading-icon]',
```
Comment (damienwebdev) [inline apps/foundation-demo/src/app/core/loading-icon/loading-icon/loading-icon.component.ts]: Can we also do `<loading-icon>` as opposed to `<div loading-icon></div>` I don't see a reason for a HostComponent vs. custom component?

File: apps/foundation-demo/src/app/core/loading-icon/loading-icon/loading-icon.component.ts (hunk 1 above)
Comment (lderrickable) [reply apps/foundation-demo/src/app/core/loading-icon/loading-icon/loading-icon.component.ts]: done

File: apps/foundation-demo/src/app/core/loading-icon/loading-icon/loading-icon.component.ts (hunk 2 above)
Comment (lderrickable) [reply apps/foundation-demo/src/app/core/loading-icon/loading-icon/loading-icon.component.ts]: done

**Pull request merged**